Description: Fix infobar visibility
 Fix vanishing infobar by periodically checking its visibility state and setting it to correct value

Index: lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.c
===================================================================
--- lightdm-gtk-greeter-1.8.5.orig/src/lightdm-gtk-greeter.c
+++ lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.c
@@ -112,6 +112,9 @@ static gboolean prompt_active = FALSE, p
 static GdkRegion *window_region = NULL;
 #endif
 
+/* Infobar visibility state */
+static gboolean infobar_is_visible = FALSE;
+
 typedef struct
 {
   gboolean is_prompt;
@@ -144,6 +147,19 @@ WindowPosition main_window_pos;
 GdkPixbuf* default_user_pixbuf = NULL;
 gchar* default_user_icon = "avatar-default";
 
+
+static gboolean
+check_infobar_visibility(gpointer data)
+{
+    if(infobar_is_visible)
+    {
+        gtk_widget_set_visible (GTK_WIDGET (info_bar), TRUE);
+    }
+
+    return TRUE;
+}
+
+
 static void
 pam_message_finalize (PAMConversationMessage *message)
 {
@@ -674,7 +690,11 @@ set_message_label (const gchar *text)
 {
     syslog (LOG_DEBUG, "[set_message_label] New message is \"%s\"", text);
     syslog (LOG_DEBUG, "[set_message_label] Info bar visibility should be %d", g_strcmp0 (text, "") != 0);
+
+    /* Set global variable about infobar visibility */
+    infobar_is_visible = g_strcmp0 (text, "") != 0;
     gtk_widget_set_visible (GTK_WIDGET (info_bar), g_strcmp0 (text, "") != 0);
+
     syslog (LOG_DEBUG, "[set_message_label] Info bar visibility is %d", gtk_widget_is_visible (GTK_WIDGET (info_bar)));
     syslog (LOG_DEBUG, "[set_message_label] Message label visibility is %d", gtk_widget_is_visible (GTK_WIDGET (message_label)));
     gtk_label_set_text (message_label, text);
@@ -2817,6 +2837,10 @@ main (int argc, char **argv)
     gdk_window_set_events (root_window, gdk_window_get_events (root_window) | GDK_SUBSTRUCTURE_MASK);
     gdk_window_add_filter (root_window, focus_upon_map, NULL);
 
+    /* Add the infobar checker here, per 200ms */
+    g_timeout_add(200, check_infobar_visibility, NULL);
+
+
 #if GTK_CHECK_VERSION (3, 0, 0)
 #else
     gdk_threads_enter();
